File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/seller/repository/SellerRepository.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.repository;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.entity.Seller;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.entity.SellerSubmitStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface SellerRepository extends JpaRepository<Seller, UUID> {

    Optional<Seller> findByUser_UserEmail(String userEmail);

    // sellerSubmitStatus가 pending인 만매자 조회
    List<Seller> findAllBySellerSubmitStatus(SellerSubmitStatus status);

    boolean existsByUser_UserId(UUID userId);

    int countBySellerSubmitStatus(SellerSubmitStatus status);
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/seller/repository/SellerRepository.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.repository

import com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.entity.Seller
import com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.entity.SellerSubmitStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface SellerRepository : JpaRepository<Seller, UUID> {
    fun findByUserUserEmail(userEmail: String): Seller?

    // sellerSubmitStatus가 pending인 만매자 조회
    fun findAllBySellerSubmitStatus(status: SellerSubmitStatus): List<Seller>

    fun existsByUserUserId(userId: UUID): Boolean

    fun countBySellerSubmitStatus(status: SellerSubmitStatus): Int
}
